transaction/builder/src/memo_builder/rth_memo_builder.rs

```diff
@@ -93,6 +124,20 @@ impl Default for RTHMemoBuilder {
     }
 }
 
+/// An error that occurs when setting up a memo builder
+///
+/// These errors are usually created setting invalid field combinations on the
+/// memo builder. We have included error codes for some known useful error
+/// conditions. For a custom MemoBuilder, you can try to reuse those, or use the
+/// Other error code.
+#[derive(Clone, Debug, Display, Eq, PartialEq)]
+pub enum MemoBuilderError {
+    /// Invalid state change
+    StateChange(String),
```

Comment:
Can we rename this to something more descriptive? I"m not sure what a `StateChange` error is...